src/core/pipeline/pipeline.js

```diff
@@ -0,0 +1,35 @@
+export default class Pipeline {
+
+	constructor(stages) {
+		this.stages = stages || [];
+	}
+
+	fetch(ctx, data = []) {
```

Comment:
why? I think promises here is overcomplicated, what is usecase for them? Maybe we can return Promise in invalidate
```javascript
qgridService.invalidate => (){
 const promise...
 const pipe = [start].concat(model.pipe).concat([finish, () => promise.resolve())
  const run = build(new PipeContext(model));
  run(pipe);
  return promise...
}
```